base/float.jl

```diff
@@ -414,9 +422,121 @@ muladd(x::T, y::T, z::T) where {T<:IEEEFloat} = muladd_float(x, y, z)
 # TODO: faster floating point fld?
 # TODO: faster floating point mod?
 
-rem(x::T, y::T) where {T<:IEEEFloat} = rem_float(x, y)
+function unbiased_exponent(x::T) where {T<:IEEEFloat}
+    exp = (reinterpret(Unsigned, x) & exponent_mask(T)) >> mantissa_width(T)
+    return exp
+end
+
+function explicit_mantissa_noinfnan(x::T) where {T<:IEEEFloat}
+    subnormal = !issubnormal(x) ? significand_mask(T) + uinttype(T)(1) : uinttype(T)(0)
+    return subnormal | (significand_mask(T) & reinterpret(Unsigned, x))
+end
+
+function make_value(number::T, ep) where {T<:Unsigned}
+    Tfloat = floattype(T)
+    Tint = signed(T)
+    epint = unsafe_trunc(Tint,ep)
+    lz::signed(T) = Core.Intrinsics.ctlz_int(number) - exponent_width(Tfloat)
+    number <<= lz
+    epint -= lz
+    bits = T(0)
+    if epint >= T(0)
+        mant = number & significand_mask(Tfloat) #this might not be necessary
+        bits |= mant
+        epint = ((epint + Tint(1)) << mantissa_width(Tfloat)) & exponent_mask(Tfloat)
+        bits |= epint
+    else
+        number = number >> -epint
+        mant = number & significand_mask(Tfloat)
+        bits |= mant
```

Comment:
```suggestion
    if epint >= T(0)
        bits = number & significand_mask(Tfloat)
        bits |= ((epint + Tint(1)) << mantissa_width(Tfloat)) & exponent_mask(Tfloat)
    else
        bits = (number >> -epint) & significand_mask(Tfloat)
```
Rather unfortunate that you can't reuse `mantissa` here since `number` is unsigned rather than a float.